blockchain/proxy_test.go

```diff
@@ -41,75 +43,132 @@ func (m *mockHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	t.Parallel()
 
-		rw.WriteHeader(http.StatusOK)
-		_, _ = rw.Write(nil)
-	}))
-	args := ArgsElrondProxy{
-		ProxyURL:       testHttpServer.URL,
-		Client:         &http.Client{},
-		SameScState:    false,
-		ShouldBeSynced: false,
-	}
-	proxy := NewElrondProxy(args)
+	t.Run("invalid time cache should error", func(t *testing.T) {
+		t.Parallel()
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*100)
-	defer cancel()
+		args := createMockArgsElrondProxy(nil)
+		args.CacheExpirationTime = time.Second - time.Nanosecond
+		proxy, err := NewElrondProxy(args)
 
-	resp, err := proxy.GetHTTP(ctx, "endpoint")
-	assert.Nil(t, resp)
-	require.NotNil(t, err)
-	assert.Equal(t, "*url.Error", fmt.Sprintf("%T", err))
-	assert.True(t, strings.Contains(err.Error(), "context deadline exceeded"))
-}
+		assert.True(t, check.IfNil(proxy))
+		assert.True(t, errors.Is(err, ErrInvalidCacherDuration))
+	})
+	t.Run("invalid nonce delta should error", func(t *testing.T) {
+		t.Parallel()
 
-func TestElrondProxy_PostHTTPContextDone(t *testing.T) {
-	t.Parallel()
+		args := createMockArgsElrondProxy(nil)
+		args.FinalityCheck = true
+		args.AllowedDeltaToFinal = 0
+		proxy, err := NewElrondProxy(args)
 
-	testHttpServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		// simulating that the operation takes a lot of time
+		assert.True(t, check.IfNil(proxy))
+		assert.True(t, errors.Is(err, ErrInvalidAllowedDeltaToFinal))
+	})
+	t.Run("should work with finality check", func(t *testing.T) {
+		t.Parallel()
 
-		time.Sleep(time.Second * 2)
+		args := createMockArgsElrondProxy(nil)
+		args.FinalityCheck = true
+		proxy, err := NewElrondProxy(args)
 
-		rw.WriteHeader(http.StatusOK)
-		_, _ = rw.Write(nil)
-	}))
-	args := ArgsElrondProxy{
-		ProxyURL:       testHttpServer.URL,
-		Client:         &http.Client{},
-		SameScState:    false,
-		ShouldBeSynced: false,
-	}
-	proxy := NewElrondProxy(args)
+		assert.False(t, check.IfNil(proxy))
+		assert.Nil(t, err)
+	})
+	t.Run("should work without finality check", func(t *testing.T) {
+		t.Parallel()
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*100)
-	defer cancel()
+		args := createMockArgsElrondProxy(nil)
+		proxy, err := NewElrondProxy(args)
 
-	resp, err := proxy.PostHTTP(ctx, "endpoint", nil)
-	assert.Nil(t, resp)
-	require.NotNil(t, err)
-	assert.Equal(t, "*url.Error", fmt.Sprintf("%T", err))
-	assert.True(t, strings.Contains(err.Error(), "context deadline exceeded"))
+		assert.False(t, check.IfNil(proxy))
+		assert.Nil(t, err)
+	})
 }
 
 func TestGetAccount(t *testing.T) {
 	t.Parallel()
 
 	httpClient := &mockHTTPClient{}
-	args := ArgsElrondProxy{
-		ProxyURL:       testHttpURL,
-		Client:         httpClient,
-		SameScState:    false,
-		ShouldBeSynced: false,
-	}
-	proxy := NewElrondProxy(args)
+	args := createMockArgsElrondProxy(httpClient)
+	proxy, err := NewElrondProxy(args)
+	require.Nil(t, err)
```

Comment:
This err is nil check is not really mandatory, since we checked in the above tests. It can also remain like that since is just 1 extra line for some tests.

Reply:
done